src/ApiGateways/Web.Bff.Shopping/aggregator/Infrastructure/HttpClientAuthorizationDelegatingHandler.cs

```diff
@@ -1,49 +1,40 @@
-﻿using Microsoft.AspNetCore.Authentication;
-using Microsoft.AspNetCore.Http;
-using System.Collections.Generic;
-using System.Net.Http;
-using System.Net.Http.Headers;
-using System.Threading;
-using System.Threading.Tasks;
-
-namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Infrastructure
+﻿namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Infrastructure;
+
+public class HttpClientAuthorizationDelegatingHandler
+        : DelegatingHandler
 {
-    public class HttpClientAuthorizationDelegatingHandler
-         : DelegatingHandler
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public HttpClientAuthorizationDelegatingHandler(IHttpContextAccessor httpContextAccessor)
     {
-        private readonly IHttpContextAccessor _httpContextAccessor;
+        _httpContextAccessor = httpContextAccessor;
+    }
 
-        public HttpClientAuthorizationDelegatingHandler(IHttpContextAccessor httpContextAccessor)
-        {
-            _httpContextAccessor = httpContextAccessor;
-        }
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        var authorizationHeader = _httpContextAccessor.HttpContext
+            .Request.Headers["Authorization"];
 
-        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        if (!string.IsNullOrEmpty(authorizationHeader))
```

Comment:
```suggestion
        if (!string.IsNullOrWhitespace(authorizationHeader))
```

The `" "` value would not be a valid auth header.

Reply:
Right.